File: app/src/main/java/xyz/godi/budgetmanager/data/TransactionDAO.java
package xyz.godi.budgetmanager.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.Date;
import java.util.List;

import xyz.godi.budgetmanager.model.Transaction;

@Dao
public interface TransactionDAO {

    // insert a transaction into the database
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Transaction transaction);

    @Query("DELETE FROM transaction_table WHERE id = :id")
    void delete(int id);

    @Query("SELECT * FROM transaction_table ORDER BY date")
    List<Transaction> getAllTransactions();

    // get transactions between two dates
    @Query("SELECT * FROM transaction_table WHERE date BETWEEN :from AND :to")
    List<Transaction> getTransactionsBetweenDates(Date from, Date to);
}

File: app/src/main/java/xyz/godi/budgetmanager/data/TransactionDao.kt
package xyz.godi.budgetmanager.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Query
import java.util.*

@Dao
interface TransactionDao {

    @Query("SELECT * FROM `transaction` ORDER BY name")
    fun getTransactions(): LiveData<List<Transaction>>

    // get transactions between dates
    @Query("SELECT * FROM `transaction` WHERE transactionDate BETWEEN :from AND :to")
    fun getTransactionsBtwnDates(from: Date, to: Date): LiveData<List<Transaction>>

    @Delete
    fun deleteTransaction(transaction: Transaction)
}